src/components/dataset/schemaCreation/DatasetSchemaBuilderView.tsx

```diff
@@ -0,0 +1,803 @@
+  Circle,
+} from '@mui/icons-material';
+import { TdrState } from 'reducers';
+import { useFormContext } from 'react-hook-form';
+import CodeMirror from '@uiw/react-codemirror';
+import { okaidia } from '@uiw/codemirror-theme-okaidia';
+import { javascript } from '@codemirror/lang-javascript';
+import {
+  ColumnModel,
+  TableDataType,
+  DatasetSpecificationModel,
+  TableModel,
+  RelationshipModel,
+} from 'generated/tdr';
+import clsx from 'clsx';
+import DatasetSchemaRelationshipModal, { wrapRadioValue } from './DatasetSchemaRelationshipModal';
+import { styles as DatasetCreationStyles } from './DatasetSchemaCommon';
+
+const styles = (theme: CustomTheme) =>
+  ({
+    ...DatasetCreationStyles(theme),
+    schemaSectionHeader: {
+      marginTop: 40,
+      paddingBottom: 5,
+      borderBottom: `2px solid ${theme.palette.terra.green}`,
+      display: 'flex',
+      alignItems: 'center',
+      fontSize: '1.5rem',
+    },
+    schemaSectionHeaderDesc: {
+      marginLeft: 10,
+    },
+    section: {
+      margin: '20px 0 10px',
+    },
+    schemaBuilderContainer: {
+      backgroundColor: theme.palette.primary.focus,
+      padding: 20,
+      marginTop: 20,
+      borderRadius: theme.shape.borderRadius,
+      marginRight: -430,
+    },
+    schemaBuilderMain: {
+      display: 'flex',
+      alignItems: 'flex-start',
+      justifyContent: 'space-between',
+    },
+    schemaBuilderStructureView: {
+      width: '55%',
+      flexShrink: 0,
+    },
+    schemaBuilderStructureViewControls: {
+      display: 'flex',
+      alignItems: 'center',
+      justifyContent: 'space-between',
+      marginTop: 5,
+    },
+    schemaControlButton: {
+      'text-transform': 'none',
+      padding: '10px 14px 10px 11px',
+      marginRight: 15,
+    },
+    iconInButton: {
+      marginRight: 6,
+    },
+    schemaBuilderDetailView: {
+      width: '100%',
+      padding: 10,
+      marginLeft: 10,
+      marginTop: 50,
+    },
+    formInputDatatype: {
+      maxWidth: 200,
+    },
+    jsonWrapper: {
+      backgroundColor: '#272822',
+      borderRadius: 10,
+      padding: 20,
+    },
+  } as any);
+
+interface IProps extends WithStyles<typeof styles> {
+  userEmail: string;
+}
+
+const DatasetSchemaBuilderView = withStyles(styles)(({ classes }: IProps) => {
+  const { register, getValues, setValue } = useFormContext();
+  const [datasetSchema, setDatasetSchema] = useState(
+    getValues().schema as DatasetSpecificationModel,
+  );
+  const [selectedTable, setSelectedTable] = useState(-1);
+  const [selectedColumn, setSelectedColumn] = useState(-1);
+  const [expandedTables, setExpandedTables] = useState({} as any);
+  const [relationshipModalOpen, setRelationshipModalOpen] = useState(false);
+  const [relationshipModalDefaultValues, setRelationshipModalDefaultValues] = useState({
+    from: '',
+    to: '',
+    expandedTables: {},
+  });
+
+  const [anchorElDetailsMenu, setAnchorElDetailsMenu] = React.useState<null | HTMLElement>(null);
```

Comment:
Does this work if you delete the null from the function call? I would expect it to default to undefined in that case.

Reply:
Probably the same issue as above, this sounds good.